device/tests/Microsoft.Azure.Devices.Client.Test/DeviceClientTests.cs

```diff
@@ -27,6 +27,35 @@ public void DeviceClient_OperationTimeoutInMilliseconds_Property_DefaultValue()
         }
 
         [TestMethod]
+        [TestCategory("IoTHubClientDiagnostic")]
+        public void DeviceClient_SetDiagnosticSamplingPercentage()
+        {
+            DeviceClient deviceClient = DeviceClient.CreateFromConnectionString(fakeConnectionString);
+            const int DefaultPercentage = 0;
+            const int ValidPercentage = 80;
+            const int InvalidPercentage = 200;
+
+            Assert.AreEqual(deviceClient.DiagnosticSamplingPercentage, DefaultPercentage);
+
+            deviceClient.DiagnosticSamplingPercentage = ValidPercentage;
+            Assert.AreEqual(deviceClient.DiagnosticSamplingPercentage, ValidPercentage);
+
+            try
+            {
+                deviceClient.DiagnosticSamplingPercentage = InvalidPercentage;
+                Assert.Fail();
+            }
+            catch (ArgumentOutOfRangeException e)
+            {
+                Assert.AreEqual(deviceClient.DiagnosticSamplingPercentage, ValidPercentage);
+            }
+            catch (Exception)
```

Comment:
You don't need to catch the exception and assert - the test runner will handle that case and fail the test. Please remove this block.

Reply:
fixed